fcrepo-stats-impl/src/main/java/org/fcrepo/stats/impl/DbStatsImpl.java

```diff
@@ -0,0 +1,113 @@
+/*
+ * The contents of this file are subject to the license and copyright
+ * detailed in the LICENSE and NOTICE files at the root of the source
+ * tree.
+ */
+package org.fcrepo.stats.impl;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.atomic.AtomicLong;
+
+import javax.annotation.PostConstruct;
+import javax.inject.Inject;
+import javax.sql.DataSource;
+
+import org.fcrepo.kernel.api.RdfLexicon;
+import org.fcrepo.stats.api.BinaryStatsResult;
+import org.fcrepo.stats.api.MimetypeStatsResult;
+import org.fcrepo.stats.api.Stats;
+import org.fcrepo.stats.api.StatsParameters;
+import org.fcrepo.stats.api.StatsResult;
+import org.fcrepo.stats.api.StatsResults;
+import org.springframework.jdbc.core.namedparam.MapSqlParameterSource;
+import org.springframework.jdbc.core.namedparam.NamedParameterJdbcTemplate;
+import org.springframework.jdbc.support.rowset.SqlRowSet;
+import org.springframework.stereotype.Component;
+import org.springframework.util.CollectionUtils;
+
+/**
+ * A database-backed implementation of the <code>Stats</code> interface.
+ * It depends on the tables that drive the search index.
+ *
+ * @author dbernstein
+ */
+@Component("stats")
+public class DbStatsImpl implements Stats {
+
+    public static final String SELECT_COUNT_FROM_SIMPLE_SEARCH = "select count(*) from simple_search";
+    @Inject
+    private DataSource dataSource;
+
+    private NamedParameterJdbcTemplate jdbcTemplate;
+
+    /**
+     * Setup template
+     */
+    @PostConstruct
+    public void setup() {
+        this.jdbcTemplate = new NamedParameterJdbcTemplate(this.dataSource);
+    }
+
+    @Override
+    public StatsResults getStatistics(final StatsParameters statsParams) {
+        final var results = new StatsResults();
+        //count all resources
+        final var resourceCount = jdbcTemplate.queryForObject(SELECT_COUNT_FROM_SIMPLE_SEARCH,
```

Comment:
I'd be curious what others think, but I find it a bit weird that the total count doesn't adjust based on my query parameters. I know the only filter at the moment is mimetype, so it'd just be the same as the binary count when the filters are active, but I still think it would be my preference to adjust the number.